jetbrains-core/src/software/aws/toolkits/jetbrains/utils/execution/steps/StepExecutor.kt

```diff
@@ -67,23 +67,25 @@ class StepExecutor(
         processHandler: DummyProcessHandler
     ) {
         try {
-            executionStarted(descriptor, progressListener)
+            executionStarted(descriptor, progressListener, processHandler)
             workflow.run(context, messageEmitter)
-            LOG.tryOrNull("Failed to invoke success callback") {
-                onSuccess?.invoke(context)
+
+            // If the dummy process was cancelled (or any step got cancelled), we need to pass on a ProcessCanceledException
+            if (context.isCancelled()) {
+                throw ProcessCanceledException()
             }
+            onSuccess?.invoke(context)
             executionFinishedSuccessfully(processHandler, progressListener)
         } catch (e: Throwable) {
```

Comment:
Should we also catch `ProcessCanceledException` explicitly and rethrow it in that case?

Reply:
We still want it to be caught and handled by the error handler so things like run configs can be shut down correctly.

Reply:
Oh I see how this works now I guess it's ok

Reply:
ya, can't make a late init for a Boolean (TIL) so needed a value.